File: Branches/VBInicial/vpPriV100GrupoMundifios/MyTools/Compras/EditorCompras/CmpIsEditorCompras.vb
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports Primavera.Extensibility.Purchases.Editors
Imports Primavera.Extensibility.Attributes
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico
Imports StdBE100

Namespace MyTools
    Public Class CmpIsEditorCompras
        Inherits EditorCompras

        Public Overrides Sub DepoisDeGravar(Filial As String, Tipo As String, Serie As String, NumDoc As Integer, e As ExtensibilityEventArgs)
            MyBase.DepoisDeGravar(Filial, Tipo, Serie, NumDoc, e)

            If Module1.VerificaToken("MyTools") = 1 Then


                '####################################################################################################################################
                '#Atualizar o CDU_DataUltimaAtualizacao apos gravaçao de uma ECFBRUNO - 04/03/2020                                                  #
                '####################################################################################################################################
                If Me.DocumentoCompra.Tipodoc = "ECF" Then


                    For i = 1 To Me.DocumentoCompra.Linhas.NumItens
                        If Me.DocumentoCompra.Linhas.GetEdita(i).Artigo & "" <> "" Then
                            Me.DocumentoCompra.Linhas.GetEdita(i).CamposUtil("CDU_DataUltimaAtualizacao").Valor = Format(Now, "yyyy-MM-dd HH:mm:ss")
                        End If
                    Next i


                End If
                '####################################################################################################################################
                '#Atualizar o CDU_DataUltimaAtualizacao apos gravaçao de uma ECF BRUNO - 04/03/2020                                                 #
                '####################################################################################################################################

            End If
        End Sub

    End Class
End Namespace

File: Trunk/vpPriV100GrupoMundifios/MyTools/Compras/EditorCompras/CmpIsEditorCompras.cs
using Generico;
using Microsoft.VisualBasic;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Purchases.Editors;
using System;

namespace MyTools
{
    public class CmpIsEditorCompras : EditorCompras
    {
        public override void DepoisDeGravar(string Filial, string Tipo, string Serie, int NumDoc, ExtensibilityEventArgs e)
        {
            base.DepoisDeGravar(Filial, Tipo, Serie, NumDoc, e);

            if (Module1.VerificaToken("MyTools") == 1)
            {
                // ####################################################################################################################################
                // #Atualizar o CDU_DataUltimaAtualizacao apos gravaçao de uma ECFBRUNO - 04/03/2020                                                  #
                // ####################################################################################################################################
                if (this.DocumentoCompra.Tipodoc == "ECF")
                {
                    for (var i = 1; i <= this.DocumentoCompra.Linhas.NumItens; i++)
                    {
                        if (this.DocumentoCompra.Linhas.GetEdita(i).Artigo + "" != "")
                            this.DocumentoCompra.Linhas.GetEdita(i).CamposUtil["CDU_DataUltimaAtualizacao"].Valor = Strings.Format(DateTime.Now, "yyyy-MM-dd HH:mm:ss");
                    }
                }
            }
        }
    }
}